src/Database/LSMTree/Internal/Index.hs

```diff
@@ -33,6 +48,21 @@ class Index i where
     -- | Yields the number of pages covered by an index.
     sizeInPages :: i -> NumPages
 
+    {-|
+        Yields the header of the serialised form of an index.
+
+        See the documentation of the 'Index' class for how to generate a
+        complete serialised index.
+    -}
+    headerLBS :: Proxy i -> LazyByteString
```

Comment:
It's a tiny optimisation, so not necessary to include, but I think it's nice to mention: [`Proxy#`](https://hackage.haskell.org/package/base-4.20.0.1/docs/GHC-Base.html#t:Proxy-35-) has no run-time representation, while `Proxy` does have run-time overhead

Reply:
Changed in 0b36ff383a6753c35fd245079142a9cd8bf74124.